### DISCUSSION TITLE
How can I flag a script as been executed already?

### DISCUSSION THREAD

**Discussion by danielo515** [twpayne/chezmoi#2282] [Q&A] [answered]
Hello.
The documentation says how to clear the state of run_once scripts, so you can re-execute them:
https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#clear-the-state-of-run_once_-scripts
However, I want exactly the opposite, to set it as already run, because i ran it manually and now I am just configuring it for other machines, however, everytime I run chezmoi apply it tries to execute it.
Thanks

**Comment by twpayne** [comment, marked as answer]
This is tricky, but possible.

First, the correct fix is to make all of your scripts, including your `run_once_` scripts,  idempotent, so it doesn't matter if they run one more than once.

If that is not possible, then you need to create an entry for the script in chezmoi's state. You will need to calculate the SHA256 sum of the script's contents (after template execution) and add it to chezmoi's state.

Say your script is called `script.sh`, you would do something like:

```console
$ sha256sum script.sh
91ee34a755e613fc83be1e399b024d1dcbd8ce22800fdb004071518e85f66945  script.sh
$ chezmoi state set --bucket=scriptState --key=91ee34a755e613fc83be1e399b024d1dcbd8ce22800fdb004071518e85f66945 --value={}
```

**Reply by danielo515** [reply]
> First, the correct fix is to make all of your scripts, including your `run_once_` scripts, idempotent, so it doesn't matter if they run one more than once.

Yes, that's the ideal scenario, however in this case is tricky, but I'm open to any suggestion. I am using a Brewfile to install all my mac-os dependencies by running `brew bundle --source /path/to/Brewfile`. The problem is that this command itself is (sadly) not idempotent, and it fails if it can not install the packages, even if the packages are already installed.
Thank you very much for your snippet.

**Reply by twpayne** [reply]
For Homebrew, I think you can use `brew bundle check` in your script to check if all dependencies are installed and only run `brew bundle` if needed.
